common/src/main/kotlin/de/hpi/cloud/common/serializers/EnumSerializer.kt

```diff
@@ -0,0 +1,22 @@
+package de.hpi.cloud.common.serializers
+
+import kotlinx.serialization.*
+import kotlinx.serialization.internal.StringDescriptor
+import kotlin.reflect.KClass
+
+abstract class EnumSerializer<E : Enum<E>>(
+    private val kClass: KClass<E>,
+    private val fallback: E,
+    private val valueToString: (E) -> String = { it.name.toLowerCase() }
+) : KSerializer<E> {
+    override val descriptor: SerialDescriptor = StringDescriptor
+
+    override fun serialize(encoder: Encoder, obj: E) = encoder.encodeString(valueToString(obj))
+
+    override fun deserialize(decoder: Decoder): E {
+        val value = decoder.decodeString()
+        return kClass.enumMembers()
+            .firstOrNull { valueToString(it).equals(value, ignoreCase = true) }
+            ?: fallback
```

Comment:
fallback could be a lambda for cases where you want to throw an error

Reply:
For a future enum this might make sense, but currently, we don't need this functionality.